packages/cli/src/lib/builder/config.ts

```diff
@@ -100,6 +100,8 @@ export async function makeRollupConfigs(
         chunkFileNames: `esm/[name]-[hash].esm.js`,
         format: 'module',
         sourcemap: true,
+        preserveModules: true,
+        preserveModulesRoot: `${targetDir}/src`,
```

Comment:
Does it work with just `'src'` here?

Reply:
Ah shoot, that's actually unearthed a larger issue, which is that those relative imports will in many cases be broken for people using our packages. Due to the layout in `node_modules` there's no guarantee that the target packages will actually exist, and more generally it's just a bad pattern to have internal cross-package imports in `node_modules`.

We'll need to either have those imports inlined somehow, or work to remove all of them.

Reply:
That is what is happening :)
Before this change, code is inlined to index.esm.js (or async fragment).
After this change, it will inline it to a new directory, that matches directory of the referenced package
![image](https://github.com/backstage/backstage/assets/603853/af33c6d0-89c8-4236-b1c7-5f036f3a1158)

Reply:
A bit ugly, but having those relative monorepo imports is ugly either way...

In the end, not breakages and the end result is the same. "Just" in a few more files and directories.